internal/weldr/api.go

```diff
@@ -1878,6 +1910,23 @@ func (api *API) composeHandler(writer http.ResponseWriter, request *http.Request
 		return
 	}
 
+	// Fetch parent ostree commit from ref + url if commit is not
+	// provided. The parameter name "parent" is perhaps slightly misleading
+	// as it represent whatever commit sha the image type requires, not
+	// strictly speaking just the parent commit.
+	if cr.OSTree.Ref != "" && cr.OSTree.URL != "" && cr.OSTree.Parent == "" {
+		parent, err := ostreeResolveRef(cr.OSTree.URL, cr.OSTree.Ref)
+		if err != nil {
+			errors := responseError{
+				ID:  "OSTree commit parsing error",
+				Msg: err.Error(),
+			}
+			statusResponseError(writer, http.StatusInternalServerError, errors)
+			return
+		}
+		cr.OSTree.Parent = parent
+	}
+
```

Comment:
Do you want to check for url and parent both set and return an error? That might prevent some confusing results if the user sets both.

Reply:
That's a good point, thanks Brian!